api/src/tool/console/interactive.py

```diff
@@ -0,0 +1,140 @@
+#
+# Interactive console for using database and services.
+#
+# Connects to the database then drops into a Python interactive prompt.
+#
+# Use via the `-i` flag to python, for example:
+#   poetry run python3 -i -m src.tool.console.interactive
+#
+
+from types import ModuleType
+
+import rich
+import rich.panel
+import rich.pretty
+
+import src.adapters.db as db
+import src.db
+import src.db.models
+import src.logging
+import src.util
+import tests.src.db.models.factories
+from src.adapters.db.clients.postgres_client import PostgresDBClient
+from src.adapters.db.clients.postgres_config import get_db_config
+from tests.src.db.models.factories import (
+    OpportunityAssistanceListingFactory,
+    OpportunityFactory,
+    OpportunitySummaryFactory,
+)
+
+INTRO = """
+Simpler Grants Gov Python console
+
+Useful things:
+
+│ db = database session
+│ f = DB factories module
+│ u = utilities module
+│ r = function to reload REPL
+
+Tips:
+★ Tab-completion is available
+★ History is available (use ↑↓ keys)
+★ Use Ctrl+D to exit
+"""
+
+
+def interactive_console() -> dict:
+    """Set up variables and print a introduction message for the interactive console."""
+
+    db_session = connect_to_database()
+
+    print(INTRO.format(**locals()))
+
+    variables = dict()
+    variables.update(vars(src.db.models.opportunity_models))
+    variables.update(vars(src.db.models.lookup_models))
+
+    # This goes after the imports of entire modules, so the console reserved
+    # names (db, fineos, etc) take precedence. This might break some modules
+    # that expect something different under those names.
+    variables.update(locals())
+
+    # DB
+    variables["db_session"] = db_session
+    variables["dbs"] = db_session
+
+    # DB Factories
+    factories_module = tests.src.db.models.factories
+    if isinstance(db_session, db.Session):
+        factories_module._db_session = db_session
+    variables["f"] = tests.src.db.models.factories
+    variables["OpportunityFactory"] = OpportunityFactory
+    variables["OpportunitySummaryFactory"] = OpportunitySummaryFactory
+    variables["OpportunityAssistanceListingFactory"] = OpportunityAssistanceListingFactory
```

Comment:
We have these all available on `f.WhateverFactory` already. Unless we want to add all of them like this (which is a lot to maintain) I'd prefer we just keep these in the `f` variable

Reply:
It was more for typing convenience for some of the main models - but sure let's stick to `f` for now